include/power/pm_policy.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_POWER_PM_POLICY_H_
+#define ZEPHYR_INCLUDE_POWER_PM_POLICY_H_
+
+#include <power/power.h>
+
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief PM Policy definition
+ *
+ * Store information about a power management policy.
+ */
+struct pm_policy {
+	enum power_states (*next_state_get)(struct pm_policy *policy,
```

Comment:
The signature has changed to add an error, but as a parameter.  For consistency with other API this should return an `int` success/error code, and store the next state as an output parameter.